rpcs3/Emu/CPU/CPUThread.cpp

```diff
@@ -192,15 +192,15 @@ struct cpu_prof
 			// Sample active threads
 			for (auto& [id, info] : threads)
 			{
-				if (auto ptr = info.wptr.lock())
+				if (cpu_flag::exit - info.ptr->state)
 				{
 					// Get short function hash
-					const u64 name = atomic_storage<u64>::load(ptr->block_hash);
+					const u64 name = atomic_storage<u64>::load(info.ptr->block_hash);
 
 					// Append occurrence
 					info.samples++;
 
-					if (!(ptr->state.load() & (cpu_flag::wait + cpu_flag::stop + cpu_flag::dbg_global_pause)))
+					if ((cpu_flag::wait + cpu_flag::stop + cpu_flag::dbg_global_pause) - info.ptr->state)
```

Comment:
I think this logic doesn't work with multiple flags, it becomes different thing

Reply:
Fixed.